### ISSUE TITLE
Executor: dynamically changing the concurrency of the Scan operator 

### ISSUE THREAD

**Issue by fzhedu** [pingcap/tidb#18448]
## Feature Request

**Is your feature request related to a problem? Please describe:**

https://github.com/pingcap/tidb/issues/18444, after increasing the concurrency of operators but decreasing the concurrency of Scan, the OOM disappears. 

**Describe the feature you'd like:**

dynamically channging the concurrency of the scan operator, because the default 15 concurrency would buffer so many regions that occupy a lot of memory. This feature would avoid OOM in many cases. 

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by SunRunAway** [comment]
Does https://github.com/pingcap/tidb/pull/17324 fix this issue?

**Comment by fzhedu** [comment]
> Does #17324 fix this issue?

no. I planned let the PR fix this issue, but let it as the TODO work. It should not open too many copWorkers at the beginning.
